pkg/api/api.go

```diff
@@ -51,13 +53,17 @@ func New(profileCatalog Catalog, logger logr.Logger) API {
 // ProfilesHandler is the handler for /profiles requests.
 func (a *API) ProfilesHandler(w http.ResponseWriter, r *http.Request) {
 	query := r.URL.Query().Get("name")
+	var result []profilesv1.ProfileCatalogEntry
 	logger := a.logger.WithValues("endpoint", r.URL.Path, "name", query)
+
 	if query == "" {
-		a.logger.Error(fmt.Errorf("missing query param"), "name param not set", "name", query)
-		a.logAndWriteHeader(w, http.StatusBadRequest)
-		return
+		logger.Info("Searching for all available profiles")
+		result = a.profileCatalog.SearchAll()
+	} else {
+		logger.Info("Searching for profiles with the name: " + query)
```

Comment:
the logger here uses structured logging, so `logger.Info("message", key, value). So a nice refactor would be:
```suggestion
		logger.Info("Searching for profiles matching name", "name", query)
```

Reply:
oops my bad, fixed